lib/class-wp-rest-widget-updater-controller.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Start: Include for phase 2
+ * Widget Updater REST API: WP_REST_Widget_Updater_Controller class
+ *
+ * @package gutenberg
+ * @since 5.2.0
+ */
+
+/**
+ * Controller which provides REST endpoint for updating a widget.
+ *
+ * @since 5.2.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Widget_Updater_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructs the controller.
+	 *
+	 * @access public
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'widgets';
+	}
+
+	/**
+	 * Registers the necessary REST API route.
+	 *
+	 * @access public
+	 */
+	public function register_routes() {
+			register_rest_route(
+				$this->namespace,
+				// Regex representing a PHP class extracted from http://php.net/manual/en/language.oop5.basic.php.
+				'/' . $this->rest_base . '/(?P<identifier>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)/',
+				array(
+					'args' => array(
+						'identifier' => array(
+							'description' => __( 'Class name of the widget.', 'gutenberg' ),
+							'type'        => 'string',
+						),
+					),
+					array(
+						'methods'             => WP_REST_Server::EDITABLE,
```

Comment:
Does this method make any changes on the server that are visible to future requests? If not, it's considered [_safe_](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Safe_methods) and is more of a `GET` request than a `POST` request.

Reply:
Initially, I used EDDITABLE because we may have a widget that has side effects (that's unexpected but we never know what plugins will do). With the improvements being made in https://github.com/WordPress/gutenberg/pull/14395, for callback widgets now we apply a change so it makes sense to use editable.